org.adempiere.base/src/org/compiere/acct/Doc_AllocationHdr.java

```diff
@@ -835,7 +835,7 @@ private String createInvoiceGainLoss (DocLine line, MAcctSchema as, Fact fact, M
 		// For Invoice
 		List<Object> valuesInv = DB.getSQLValueObjectsEx(getTrxName(), sql.toString(),
 				MInvoice.Table_ID, invoice.getC_Invoice_ID(), as.getC_AcctSchema_ID(), acct.getAccount_ID());
-		if (valuesInv != null) {
+		if (valuesInv != null && valuesInv.size() >= 4) {
```

Comment:
why need to add (valuesInv.size() >= 4)
size of valuesInv depend on columns on sql, so always = 4 until change sql